### PR TITLE
aswanth

### PR DESCRIPTION
**Pull request by aswanthpp**

### CODE REVIEW

File: Participants/aswanthpp.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Name: Aswanth P P
+Handle :aswanhtpp
```
Comment (rohitvarkey) [inline Participants/aswanthpp.txt]: I think you misspelled this. :)

File: binary_search/binary_search.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
+This  Pull request resolves Issue #27
```
Comment (rohitvarkey) [inline binary_search/binary_search.c]: Should be a comment!

File: binary_search/binary_search.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+#include <stdio.h>
+
+// A iterative binary search function. It returns location of x in
+// given array arr[l..r] if present, otherwise -1
+int binarySearch(int arr[], int l, int r, int x)
+{
+while (l <= r)
```
Comment (rohitvarkey) [inline binary_search/binary_search.c]: nit: indent the function block

File: binary_search/binary_search.c [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{
+	int m = l + (r-l)/2;
```
Comment (rohitvarkey) [inline binary_search/binary_search.c]: `(l + (r - l))/2`
C operator preferences can be tricky!

File: binary_search/binary_search.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		r = m - 1; 
+}
+
+// if we reach here, then element was not present
+return -1; 
+}
+
+int main(void)
+{
+int arr[] = {2, 3, 4, 10, 40};
```
Comment (rohitvarkey) [inline binary_search/binary_search.c]: nit: indent the function block

File: binary_search/binary_search.c [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+int n = sizeof(arr)/ sizeof(arr[0]);
+int x = 10;
+int result = binarySearch(arr, 0, n-1, x);
+(result == -1)? printf("Element is not present in array")
```
Comment (rohitvarkey) [inline binary_search/binary_search.c]: I think a `if` `else` block would make this more clearer.

File: binary_search/binary_search.c [hunk 7]
```diff
@@ -0,0 +1,39 @@
+This  Pull request resolves Issue #27
+
+#include <stdio.h>
+
+// A iterative binary search function. It returns location of x in
+// given array arr[l..r] if present, otherwise -1
```
Comment (rohitvarkey) [inline binary_search/binary_search.c]: Maybe also mention the array has to be sorted?

File: binary_search/binary_search.c (hunk 2 above)
Comment (aswanthpp) [reply binary_search/binary_search.c]: Ooh I just added top of  the code
This was my First PR

File: binary_search/binary_search.c (hunk 7 above)
Comment (aswanthpp) [reply binary_search/binary_search.c]: Ok I will add that also

**Comment by salman-bhai** [comment]
Any updates @aswanthpp

**Comment by salman-bhai** [comment]
Are there any updates here or should we close the PR and label it as `spam`?

**Comment by salman-bhai** [comment]
spam

**Pull request closed without merging**